codec/types/compat.go

```diff
@@ -50,7 +50,7 @@ func (any *Any) UnmarshalAmino(bz []byte) error {
 	return nil
 }
 
-func (any Any) MarshalJSON() ([]byte, error) {
+func (any *Any) MarshalJSON() ([]byte, error) {
```

Comment:
Why do we need a pointer reference when marshaling?

Reply:
Using Any, `any.compat` (just below) is `nil`. Using *Any, `any.compat` is correct. I'm not 100% I understand why; it's set on [this line](https://github.com/cosmos/cosmos-sdk/blob/d61fa431085694d6a31a8821c0a496baf4942fa0/codec/types/compat.go#L207) btw.

I didn't dig too much into it, but anyways, without the reference, tests fail.

Reply:
@amaurymartiny , this could be some weird concurrency issue. 

If you use `Any` then a method receiver works on an object copy. Sometimes it's a source of errors in methods which modifies the object. Here all methods which modifies the object are using pointer receivers. So that's why my guess is that tests are doing something weird in parallel.